Add request parameter to jwt_decode_handler (fixes #243)

MattHardcastle opened the pull request:
Currently only the `jwt_token` is passed to the `jwt_decode_handler`. This pull request adds the `request` parameter if `jwt_decode_handler` throws a `TypeError` while being called with only one parameter (which happens if `jwt_decode_handler` takes more than one parameter).

blueyed commented:
I like the idea of passing in `request`, but there might be a better way to support both function signatures?!

Also: needs to be rebased.

MattHardcastle commented:
Thanks for considering my pull request. I've rebased it with the latest master. I also changed the signature check method from catching the TypeError exception to actually inspecting the code object to see how many args it takes.

I initially wrote it to support both signature because I'm sensitive to the fact that it's a breaking change. But it's probably best to pass requests as a keyword argument and take the breaking change.

MattHardcastle commented:
I may reopen this once I get legal ok from my employer.
